app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt

```diff
@@ -30,7 +45,54 @@ class ServerErrorInterceptor(
         }
     }
 
-    private fun handleResponse(response: Response): Response {
+    private val requestBuffer = Buffer()
+    private fun createRpcRequest(chain: Interceptor.Chain): Request {
+        val request = chain.request()
+
+        val json = getRequestJson(request)
+
+        // Refactor this
+        val url = request.url
+
+        val httpUrl = if (json.getString("method") == "getConfirmedTransaction") {
+            url.newBuilder().host(getBaseUrl(Environment.RPC_POOL)).build()
+        } else {
+            url.newBuilder().host(getBaseUrl(currentEnvironment)).build()
+        }
+
+        return request.newBuilder().url(httpUrl).build()
```

Comment:
there are too many newBuilder() calls in here,
how about that:
```suggestion
        val environmentUrl = if (json.getString("method") == "getConfirmedTransaction") Environment.RPC_POOL else currentEnvironment
        
        val httpUrl = url.newBuilder()
            .host(getBaseUrl(environmentUrl))
            .build()

        return request.newBuilder()
            .url(httpUrl)
            .build()
```
```